File: hw/bonilla/u2/hw19Inheritance/FarmSystemJavaScript/model/Pig.js
const FarmAnimal = require('./FarmAnimal');

class Pig extends FarmAnimal {
    constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
    }

    toString() {
        return `{
  "farmAnimal": ${super.toString()}
}`;
    }
}

module.exports = Pig;

File: hw/hidalgo/u2/hw19Inheritance/Java/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/Pig.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 * @author Mikael Hidalgo, Object Masters, @ESPE
 */
public class Pig extends FarmAnimal {

    public Pig(int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
    }
    
    @Override
    public String toString() {
        return "Pig{" + super.toString() + super.toString()+'}';
    }
}